src/input.rs

```diff
@@ -437,6 +532,27 @@ mod tests {
         }
     }
 
+    test_clickstate! {
+        name: single_click,
+        initial_state: ClickState::None,
+        input: Event::MouseInput(ElementState::Pressed, MouseButton::Left),
+        end_state: ClickState::Click
+    }
+
+    test_clickstate! {
+        name: double_click,
+        initial_state: ClickState::Click,
+        input: Event::MouseInput(ElementState::Pressed, MouseButton::Left),
+        end_state: ClickState::DblClick
```

Comment:
I think the state transitions should still be done where they are (the processor is solely responsible for updating the Mouse state). The `ActionContext` trait advertises that the methods will be called though, so it would be nice to have coverage on that as well.